pkg/statemachine/msgbuffers.go

```diff
@@ -35,17 +36,30 @@ func (nbs *nodeBuffers) nodeBuffer(source nodeID) *nodeBuffer {
 			id:       source,
 			logger:   nbs.logger,
 			myConfig: nbs.myConfig,
+			msgBufs:  map[*msgBuffer]struct{}{},
 		}
 	}
 
 	return nb
 }
 
+func (nbs *nodeBuffers) status() []*status.NodeBuffer {
+	var stats []*status.NodeBuffer
```

Comment:
Minor issue, especially since this is in the status path, but we know the size of this slice ahead of time, so no reason not to allocate it more efficiently as:

```
stats := make([]*status.NodeBuffer, 0, len(nbs.nodeMap))
```